### ISSUE TITLE
Feature: Training & Validation Graph Display on “Train” Button Click

### ISSUE THREAD

**Issue by Jaimin020** [Jaimin020/ModelForge#11]
## Description
When the user clicks the **"Train"** button, the system should begin the training process and simultaneously display real-time graphs. These graphs will show **training and validation metrics** to help users track model performance over time.

---

### Trigger
- Clicking the **“Train”** button must:
  - Start the training process.
  - Trigger a graph component to render or update in real-time.

### Graph Metrics
The following metrics should be visualized:
- **Training Loss**
- **Validation Loss**
- **Training Accuracy**
- **Validation Accuracy**

### Graph Type
- Use **line charts** for clear visualization.
- **X-axis**: Epoch number (or batch index if applicable).
- **Y-axis**: Metric values (accuracy as % or loss as scalar).

### Graph Features
- Real-time updates during training.
- Tooltip on hover with metric values.
- Legend for curve identification.
- Zoom/pan support (optional).
- Option to pause/stop training (backend dependent).

### Layout
- Display graphs in one of the following:
  - A **"Training Dashboard"** section below the model view.
  - A **modal** if screen space is constrained.
- Arrange graphs in a 2x2 grid (for the four metrics).

### Backend Requirements
- Backend must stream metric data in JSON format:
  ```json
  {
    "epoch": 1,
    "train_loss": 0.543,
    "val_loss": 0.589,
    "train_accuracy": 0.87,
    "val_accuracy": 0.84
  }
